### ISSUE TITLE
Embedded records always sideloaded

### ISSUE THREAD

**Issue by patocallaghan** [danielspaniel/ember-data-factory-guy#183]
Taking the following Ember Data model with a `hasMany` relationship.

``` js
//app/models/message.js
export default DS.Model.extend({
  title: DS.attr('string'),
  variations: DS.hasMany('variations')
});
```

with the following serializer:

``` js
//app/serializers/message.js
export default Serializer.extend(DS.EmbeddedRecordsMixin, {
  attrs: {
    variations: { embedded: 'always' }
  }
});
```

When using `mockQuery` or the like the resulting JSON always ends up sideloaded

```
{
    variations: [...]
    messages: [...]
}
```

rather than embedded like:

```
[
    {
        "id": "1",
        "title": "Message Title",
        "variations": [
            {
                "id": "1",
                "body": "Variation 1"
            }
        ],
    },
    {
        "id": "2",
        "title": "Message Title",
        "variations": [
            {
                "id": "1",
                "body": "Variation 1"
            }
        ],
    }
]
```

**Comment by danielspaniel** [comment]
thanks .. an authorized representative will be on this asap :)

**Comment by danielspaniel** [comment]
This issue is now fixed in v2.4.3 .. thanks for the bringing of this upppppp
